.github/workflows/ci.yml

```diff
@@ -4,82 +4,101 @@ on:
   push:
     branches: [ master ]
   pull_request:
-    branches: [ master ]
 
 jobs:
-  build:
-    runs-on: ubuntu-latest
-    steps:
-    - name: PHP Installation
-      uses: shivammathur/setup-php@v2
-      with:
-        php-version: '7.4'
-        extensions: bcmath
-    - name: Code Checkout
-      uses: actions/checkout@v2
-    - name: Cache
-      id: composer-cache
-      uses: actions/cache@v2
-      with:
-        path: vendor
-        key: ${{ runner.os }}-php-${{ hashFiles('src/', 'tests/') }}
-        restore-keys: |
-          ${{ runner.os }}-php-
-    - name: Installation
-      if: steps.composer-cache.outputs.cache-hit != 'true'
-      run: composer install --prefer-dist --no-progress --no-suggest
-    - name: Prepare Artifacts
-      run: tar -cvf vendor.tar vendor
-    - name: Archive Artifacts
-      uses: actions/upload-artifact@v2
-      with:
-        name: test-dist
-        retention-days: 1
-        path: vendor.tar
-
   analysis:
-    runs-on: ubuntu-latest
-    needs: build
+    strategy:
+      matrix:
```

Comment:
Matrix is not really necessary here, but useful if you need to quickly verify any linting/analysis jobs in future, by adding a couple lines here